### ISSUE TITLE
`ConcurrentModificationException` thrown in `org.gradle.internal.serialize.DefaultSerializerRegistry`

### ISSUE THREAD

**Issue by jschear** [gradle/gradle#18152]
### Expected Behavior
No exception should be thrown.

### Current Behavior
A `java.util.ConcurrentModificationException` is thrown.

### Context
I'm running a task defined by the [SQLDelight Gradle plugin](https://github.com/cashapp/sqldelight) (using version 1.5.1). [(The task implementation is here.)](https://github.com/cashapp/sqldelight/blob/master/sqldelight-gradle-plugin/src/main/kotlin/com/squareup/sqldelight/gradle/VerifyMigrationTask.kt#L62) The task uses the Worker API, and I'm seeing a `ConcurrentModificationException` thrown by Gradle internals. I'm not fully clear on whether this is an issue in Gradle or the task's implementation. The failure is nondeterministic; it occurs in fewer than 5% of builds.

### Steps to Reproduce
I'm happy to try to reproduce this in a small project if that would be helpful. I'm hoping that the exception itself will be enough to identify the issue.

### Your Environment
I can't provide a build scan, but our project is using:
- Gradle 7.2
- SqlDelight 1.5.1

Full stacktrace:
```
[2021-08-26T19:59:21.658Z] FAILURE: Build failed with an exception.
[2021-08-26T19:59:21.658Z] 
[2021-08-26T19:59:21.658Z] * What went wrong:
[2021-08-26T19:59:21.658Z] Execution failed for task ':libraries:telemetry:telemetry-db:verifyMainTelemetryDatabaseMigration'.
[2021-08-26T19:59:21.658Z] > A failure occurred while executing com.squareup.sqldelight.gradle.VerifyMigrationTask$VerifyMigrationAction
[2021-08-26T19:59:21.658Z]    > Could not serialize unit of work.
[2021-08-26T19:59:21.658Z]       > java.util.ConcurrentModificationException (no error message)
[2021-08-26T19:59:21.658Z] 
[2021-08-26T19:59:21.658Z] * Try:
[2021-08-26T19:59:21.658Z] Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[2021-08-26T19:59:21.658Z] 
[2021-08-26T19:59:21.658Z] * Exception is:
[2021-08-26T19:59:21.658Z] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':libraries:telemetry:telemetry-db:verifyMainTelemetryDatabaseMigration'.
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:188)
[2021-08-26T19:59:21.658Z] 	at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:186)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:174)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
[2021-08-26T19:59:21.658Z] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
[2021-08-26T19:59:21.658Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
[2021-08-26T19:59:21.658Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
[2021-08-26T19:59:21.658Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
[2021-08-26T19:59:21.658Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
[2021-08-26T19:59:21.658Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
[2021-08-26T19:59:21.658Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
[2021-08-26T19:59:21.658Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
[2021-08-26T19:59:21.658Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:79)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:79)
[2021-08-26T19:59:21.659Z] 	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
[2021-08-26T19:59:21.659Z] 	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
[2021-08-26T19:59:21.659Z] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:408)
[2021-08-26T19:59:21.659Z] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:395)
[2021-08-26T19:59:21.659Z] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:388)
[2021-08-26T19:59:21.659Z] 	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:374)
[2021-08-26T19:59:21.659Z] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
[2021-08-26T19:59:21.659Z] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
[2021-08-26T19:59:21.659Z] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
[2021-08-26T19:59:21.659Z] 	at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
[2021-08-26T19:59:21.659Z] Caused by: org.gradle.workers.internal.DefaultWorkerExecutor$WorkExecutionException: A failure occurred while executing com.squareup.sqldelight.gradle.VerifyMigrationTask$VerifyMigrationAction
[2021-08-26T19:59:21.659Z] 	at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$2(DefaultWorkerExecutor.java:208)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:214)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
[2021-08-26T19:59:21.659Z] Caused by: org.gradle.workers.internal.WorkSerializationException: Could not serialize unit of work.
[2021-08-26T19:59:21.659Z] 	at org.gradle.workers.internal.DefaultActionExecutionSpecFactory.serialize(DefaultActionExecutionSpecFactory.java:79)
[2021-08-26T19:59:21.659Z] 	at org.gradle.workers.internal.DefaultActionExecutionSpecFactory.newTransportableSpec(DefaultActionExecutionSpecFactory.java:42)
[2021-08-26T19:59:21.659Z] 	at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.lambda$execute$0(IsolatedClassloaderWorkerFactory.java:51)
[2021-08-26T19:59:21.659Z] 	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
[2021-08-26T19:59:21.659Z] 	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
[2021-08-26T19:59:21.659Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:79)
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:79)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.execute(IsolatedClassloaderWorkerFactory.java:49)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$2(DefaultWorkerExecutor.java:206)
[2021-08-26T19:59:21.660Z] 	... 6 more
[2021-08-26T19:59:21.660Z] Caused by: java.util.ConcurrentModificationException
[2021-08-26T19:59:21.660Z] 	at org.gradle.internal.serialize.DefaultSerializerRegistry.build(DefaultSerializerRegistry.java:84)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeState(IsolatableSerializerRegistry.java:188)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry.access$1800(IsolatableSerializerRegistry.java:62)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedImmutableManagedValueSerializer.write(IsolatableSerializerRegistry.java:318)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedImmutableManagedValueSerializer.write(IsolatableSerializerRegistry.java:312)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:345)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:338)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:345)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:338)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatableSequence(IsolatableSerializerRegistry.java:142)
[2021-08-26T19:59:21.660Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry.access$2200(IsolatableSerializerRegistry.java:62)
[2021-08-26T19:59:21.661Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedArraySerializer.write(IsolatableSerializerRegistry.java:515)
[2021-08-26T19:59:21.661Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedArraySerializer.write(IsolatableSerializerRegistry.java:510)
[2021-08-26T19:59:21.661Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
[2021-08-26T19:59:21.661Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:345)
[2021-08-26T19:59:21.661Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:338)
[2021-08-26T19:59:21.661Z] 	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
[2021-08-26T19:59:21.661Z] 	at org.gradle.workers.internal.DefaultActionExecutionSpecFactory.serialize(DefaultActionExecutionSpecFactory.java:76)
[2021-08-26T19:59:21.661Z] 	... 23 more
```

**Comment by jbartok** [comment]
@jschear Yes, if you don't mind a small reproducer would be very helpful.

**Comment by jschear** [comment]
I'll try to put together a reproducer today or tomorrow. We also saw this on Gradle 7.1.1.

**Comment by jschear** [comment]
Unfortunately I was unable to reproduce this in a small project. (Here's the project I was trying with: 
[gradle-worker-cme-repro.zip](https://github.com/gradle/gradle/files/7100179/gradle-worker-cme-repro.zip))

**Comment by jbartok** [comment]
Thank you for giving it a try. @gradle/configuration-cache could you pls. take a look?

**Comment by KaiyuanZhao** [comment]
same issue in gradle6.7.1
we are using classLoaderIsolation workQueue

```
Cause 1: org.gradle.workers.internal.DefaultWorkerExecutor$WorkExecutionException: A failure occurred while executing ******
	at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$2(DefaultWorkerExecutor.java:202)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:215)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.gradle.workers.internal.WorkSerializationException: Could not serialize unit of work.
	at org.gradle.workers.internal.DefaultActionExecutionSpecFactory.serialize(DefaultActionExecutionSpecFactory.java:79)
	at org.gradle.workers.internal.DefaultActionExecutionSpecFactory.newTransportableSpec(DefaultActionExecutionSpecFactory.java:42)
	at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.lambda$execute$0(IsolatedClassloaderWorkerFactory.java:52)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
	at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
	at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
	at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.execute(IsolatedClassloaderWorkerFactory.java:50)
	at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$2(DefaultWorkerExecutor.java:200)
	... 12 more
Caused by: java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1445)
	at java.util.HashMap$KeyIterator.next(HashMap.java:1469)
	at org.gradle.internal.serialize.DefaultSerializerRegistry.build(DefaultSerializerRegistry.java:84)
	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeState(IsolatableSerializerRegistry.java:188)
	at org.gradle.workers.internal.IsolatableSerializerRegistry.access$1800(IsolatableSerializerRegistry.java:62)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedImmutableManagedValueSerializer.write(IsolatableSerializerRegistry.java:318)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedImmutableManagedValueSerializer.write(IsolatableSerializerRegistry.java:312)
	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:345)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:338)
	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:345)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:338)
	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatableSequence(IsolatableSerializerRegistry.java:142)
	at org.gradle.workers.internal.IsolatableSerializerRegistry.access$2200(IsolatableSerializerRegistry.java:62)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedArraySerializer.write(IsolatableSerializerRegistry.java:515)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedArraySerializer.write(IsolatableSerializerRegistry.java:510)
	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:345)
	at org.gradle.workers.internal.IsolatableSerializerRegistry$IsolatedManagedValueSerializer.write(IsolatableSerializerRegistry.java:338)
	at org.gradle.workers.internal.IsolatableSerializerRegistry.writeIsolatable(IsolatableSerializerRegistry.java:129)
	at org.gradle.workers.internal.DefaultActionExecutionSpecFactory.serialize(DefaultActionExecutionSpecFactory.java:76)
	... 29 more
```

**Comment by mlopatkin** [comment]
@gradle/bt-execution it seems to be an issue with worker isolation, could you please take a look?

**Comment by lptr** [comment]
I think this should have been fixed in Gradle 7.5 by: https://github.com/gradle/gradle/pull/20200

Can you reproduce with a recent version?

**Comment by wolfs** [comment]
I'll close this now, please comment if the problem persists.
